tests/e2e/utils.js

```diff
@@ -12,56 +12,87 @@
+ */
+
 /**
- * @fileoverview Functional components for end-to-end tests with protractor.
+ * @type {string} Log type for browser console logs.
  */
+const LOG_TYPE_BROWSER = 'browser';
+
 
 /**
- * Defines the number of logs that the log can hold.
+ * Retrieves log entries currently present in the browser console.
+ *
+ * The minimal level of captured logs is configured in protractor.conf.js.
  *
- * @type {number}
+ * Note: This should only be called once per spec as it will also clear the
+ * browser console logs. Subsequent calls will return an empty list if no
+ * logging has occurred since.
+ *
+ * @return {number}
+ */
+var getConsoleLogs = async function() {
+  return browser.manage().logs().get(LOG_TYPE_BROWSER);
+};
+
+/**
+ * Convenience function used in tests to expect no console output.
  */
-var CONSOLE_LOG_THRESHOLD = 900;
+var expectNoConsoleLogs = async function() {
+  let consoleLogs = await getConsoleLogs();
+  expect(consoleLogs.length).toBe(0);
+};
+
 /**
- * Array defining which errors one can ignore printing in the console
- * automatically. Will be concatenated with errors to ignore that will be
- * passed in.
+ * Legacy function.
+ * TODO(davyrisso): Remove when references are removed from the server version.
  *
- * @type {Array}
+ * @param {Array} errorsToIgnore A list of errors to ignore.
  */
-var CONSOLE_ERRORS_TO_IGNORE = [];
+var checkForConsoleErrors = async function(errorsToIgnore) {};
+
 
 /**
- * Manages the logs made in the browser's console to then select which ones
- * to push to the console and which ones can be ignored.
+ * Returns the current size of the browser window.
  *
- * @param {Array} errorsToIgnore
+ * @returns {Object} An object that holds the dimensions of the window.
  */
-var checkForConsoleErrors = async function(errorsToIgnore) {
-  var irrelevantErrors =
-      (errorsToIgnore || []).concat(CONSOLE_ERRORS_TO_IGNORE);
+var getWindowSize = async function() {
+  return await browser.manage().window().getSize();
+};
 
-  var browserLogs = await browser.manage().logs().get('browser');
-  var fatalErrors = [];
+/**
+ * Sets the dimensions of the browser window.
+ *
+ * @param {number} width The desired width of the window in pixels.
+ * @param {number} height The desired width of the window in pixels.
+ */
+var setWindowSize = async function(width, height) {
+  await browser.driver.manage().window().setSize(width, height);
+};
 
-  for (var i = 0; i < browserLogs.length; i++) {
-    if (browserLogs[i].level.value > CONSOLE_LOG_THRESHOLD) {
-      var errorFatal = true;
-      for (var j = 0; j < irrelevantErrors.length; j++) {
-        if (browserLogs[i].message.match(irrelevantErrors[j])) {
-          errorFatal = false;
-        }
-      }
-      if (errorFatal) {
-        fatalErrors.push(browserLogs[i]);
-      }
-    }
-  }
+/**
+ * Sets the dimensions of the browser window to those of a large screen.
+ */
+var setLargeScreen = async function() {
+  await setWindowSize(1280, 1024);
```

Comment:
@rabidbit could you please confirm that this (and below) are reasonable dimensions to be testing with? If students are going to be using a smaller screen then our e2e tests should cover that instead.

Reply:
Acknowledged, awaiting @rabidbit validation. Looking at the CSS in LearnerViewDirective.js the breakpoint is set at width <= 1058px.

Reply:
Sorry, what's the context? Is this the minimum size representing someone on a desktop with a large window (i.e., maximum width)? Is this for the stand-alone or embedded version?

Reply:
I see. If you are just testing to see if elements are re-arranged after window re-size, then I think 1070 x 700 (wide enough to see elements in two columns) to 1050 x 700 (narrow enough to see elements in one column) should work.

Reply:
Done. Thanks!